### ISSUE TITLE
self-consistent decision criteria for what effects to include in DC2

### ISSUE THREAD

**Issue by rmandelb** [LSSTDESC/DC2-production#144]
I suggest that before we finalize the list of effects included in DC2 at the final DC2 checkpoint, we apply a coherent/consistent set of decision criteria.  This comes out of a [slack conversation](https://lsstc.slack.com/archives/C77DDKZHR/p1521552916000639) (which was itself based on various conversations in this repo) specifically about effects for which there is not currently a correction in DM.

The criteria that @rmjarvis and I discussed were:

(a) the effects are small, so we can learn what they do to our science when uncorrected, but they won’t completely ruin our ability to test our pipelines in other ways for DC2. 
 
(b) the effects are traceable in some sense (one can correlate against specific quantities to identify their impact - e.g., for DCR we can correlate against airmass).  

(c) DM does plan to correct for them eventually, so we can later reprocess.  

(d) Would having a large suite of images with the effect be useful for DM in developing the appropriate correction?  If so, then it would be nice to let DM work with DC2 images rather than wait for the next time we do a huge simulation like this.  

(e) I think a corollary of (d) that is perhaps obvious (but it's better to be explicit) is this: the effect should be included in the sims in a way that is physically meaningful (based on real data and physics of LSST detectors), to ensure that the sims would actually be useful for developing and testing corrections.  If the simulation implementation doesn't reflect key features of the effect in reality, that might be used to build the correction or that might determine how important the effect is for LSST, then it's better to have it off than to have it in DC2.  (Special-order sims might still be useful but we shouldn't use that much of our resources on it.)  <-- Edited to say that I added this one later, after the initial responses in this thread.

Previous discussions focused on considerations related to (a)-(c).  For example, the argument is that tree rings and DCR meet conditions (a)-(c),  and proper motions are being excluded because of (a) (they were the dominant astrometric residual in DC1).

**Comment by rmjarvis** [comment]
Starting with the ones already mentioned:

* Tree rings.  I think yes to all.
* DCR.  Yes to a,b,c, but not d.
* Proper motion.  No to a.  Although maybe yes to d.  But unfortunately, a wins I think.

What else needs to be decided?

**Comment by rmjarvis** [comment]
* Brighter fatter. Yes to a,b,c.  Maybe d?
* Clouds.  Yes to a,b,c.  Probably d.
* Fringing.  Yes to a,c.  Not really b.  And probably not d.  (I can't remember if this is even an option for PhoSim.  It's not in ImSim.)
* Cosmic rays.  Yes to a.  Not b.  I think DM already includes this?, so c,d may not be relevant.  And I don't know if it's an option for either PhoSim or ImSim.

**Comment by rmandelb** [comment]
@rmjarvis - I decided to break out my "realism of simulation of the effect" point into a separate point (e).  Does that make sense to you?  I think it's worth considering, and probably has to be considered separately for ImSim and PhoSim. 

In the spirit of collecting info in a useful format, below is my attempt at a table of effects from your messages.  For effects that fail on one of the earlier criteria I didn't necessarily try to fill in the rest of the columns.

| Effect name | (a) | (b) | (c) | (d) | (e) PhoSim | (e) ImSim |
|-------------|----|----|----|----|------------|----------|
| Tree rings    | y  | y | y | y | single model based on ? | lab-based mixed focal-plane model |
|  DCR | y | y | y| n | we know this physics | we know this physics |
| Proper motion | n | - | - | - | - | - |
| Brighter/fatter | y | y| y| maybe? | ? | ?|
|Clouds | y | y | y | probably? | ? | not implemented |
| Fringing | y | probably not | y | probably not  | ? | not implemented |

Happy to abandon if this is not actually useful, but if people find this table helpful then feel free to edit this GitHub comment to add rows for effects that we need to discuss.

**Comment by rmjarvis** [comment]
Fancy!  I like it.

**Comment by cwwalter** [comment]
One thing we (Jim, Craig and I) would like to discuss (and maybe this is a new line for Rachel's table) is what the current desired behavior of sky background photons should be relative to tree rings.  In imSim we could choose to leave it on for the sources but completely turn it off for the sky background.  We could also have the sky photons be astrometrically shifted.  We have to work on the most efficient way to run in this mode, but first we wonder how we should approach it.

As discussed a lot in the last few days, currently DM doesn't have a way to deal with this other than a QE effect which will bias our flux measurements. OTOH these effects are there and we hope we will be able to use DC2 Run2 files for quite a while.  What are people's thoughts? @rmjarvis?

**Comment by salmanhabib** [comment]
Umm, actually the PhoSim tree-ring model is also "lab-based" in the sense that the parameters come from BNL data. Andrei has a student at BNL working on determining the PhoSim parameters. The e2v sensors don't have TRs and the ITL TRs can be modeled using PhoSim parameters; we are starting running the PhoSim flats at Argonne now and will work with BNL on the PhoSim TR models.

**Comment by rmjarvis** [comment]
Thanks. Salman.  I added a summary of that info into the table.

BTW, people keep claiming "The e2v sensors don't have TRs", which I believe just means they aren't detectable.  They must be there at some level, just below the level we've been able to look for them.

**Comment by salmanhabib** [comment]
The e2v quote is a direct statement from Andrei. I agree that what he means is that they are much lower than the ITL TRs (order of magnitude?). I'll ask him tomorrow about them.

**Comment by RobertLuptonTheGood** [comment]
> One thing we (Jim, Craig and I) would like to discuss (and maybe this is a new line for Rachel's table) is what the current desired behavior of sky background photons should be relative to tree rings. In imSim we could choose to leave it on for the sources but completely turn it off for the sky background. We could also have the sky photons be astrometrically shifted. We have to work on the most efficient way to run in this mode, but first we wonder how we should approach it.

I think all photons should be treated identically.  If we have tree rings they should be fully implemented (at a level consistent with the LSST chips when run at the mountain-top backside voltage).  I don't really think that they'll be an issue.

**Comment by kadrlica** [comment]
@rmandelb following on Mike's comment and the discussion on Slack, do we want to include CRs in your table. I agree with Mike's assessment:
(a) y (b) n (c) already corrected (d) simulations probably won't help too much (e) at the DC2 telecon it was mentioned by Simon and Jim that CRs were being included based on a template library from sensor data.
In summary, I think CRs are included and already corrected for; so no action is required. However, we may still want to track them in the table.

**Comment by kadrlica** [comment]
Based on experience with DES here are some other low-level features that have not been much discussed:
1) Scattered light from bright stars. Are reflections off of the CCDs and internal camera optics included by phosim? Should we expect to see ghosting? Does DM have any mechanism to flag/remove affected CCDs.
2) Bad pixels and columns. I haven't seen any in the test runs, but maybe they are already solved through flagging and interpolated?
3) Earth orbiting satellite streaks. I assume these are not included and we don't have a model for including them. I think that DES did measure some noticeable effects in the shears for SVA1 (@rmjarvis can comment more).
4) Fringing (especially in z,y filters). I know that some work started on including this in galsim, but I'm not sure if it ever got anywhere.

**Comment by RobertLuptonTheGood** [comment]
> Scattered light from bright stars. Are reflections off of the CCDs and internal camera optics should be included by phosim. I think this means that we expect to see ghosting. Does DM have any mechanism to flag/remove affected CCDs?

We see this in HSC, but the LSST optical design implies that there are no localised ghosts.  The code can handle it (but not very well), so we should certainly leave them it.  We do not have code to flag bad CCDs pending on sky data.

> Bad pixels and columns. I haven't seen any in the test runs, but maybe they are already correct through flagging and interpolated? Maybe they are just at a much lower level than I'm used to from DECam? Anyway, I think the plan is to include them?

Yes.  We'll either need a list, or enough simulated calibration data to find them -- the former is preferable for now.

> Cross talk. From various eyeball reports on the Run 1.1p images, cross talk is being included in the simulation. Is DM correcting this? Is the same cross talk matrix that is used to generate the simulated cross talk also being used to correct it?

We routinely take out HSC cross-talk based on coefficients measured from data (using bleed trails and/or CRs).  I'm not sure that this measurement code is automated, as we plan to use calib data such as the CBP to measure the coefficients.  So currently yes, I think that there's a danger of circularity.

DECam has much worse cross-talk than any other camera I've used;  HSC and I think LSST only has non-zero coefficients within a single CCD.  There is some prototype code to handle whole-focal-plane crosstalk, but I don't think we'll need it.

> Non-linearity at low and high light levels. I've been told that this is included in the simulations, but I don't know if/how it is being dealt with by DM. I don't know if we have a check to verify the performance.

If given a non-linearity model we'll correct it.  Again, there's a danger of circularity as phoSim is not generating the lab data used to measure the non-linearity (and I'm not sure that we have a canned way to use it, although that isn't hard)

> Fringing (especially in z,y filters). From the specs doc, fringing is turned off in run 1.1p, but from discussions it sounded like it would be turned on in the next run? I know that some work started on including fringing in galsim/imsim, but I'm not sure that it ever progressed.

Fringing is probably only visible in y.  We can handle fringing (I think only a single component is needed for HSC, which is thicker but has a slower beam).  I am concerned about coupling of the variation of the OH spectrum with time coupled to the extreme thickness variation seen in LSST chips (20 fringes centre to edge at 1 micron!), but I don't think phoSim simulates this.  There are also other strangenesses in LSST fringing (spider webs) that phoSim may not include but are probably not a problem.

> Earth orbiting satellite streaks. These are not included and I don't think they are being seriously considered. I think this is fine, but DES did measure satellite streaks having some effects in the shears for SVA1 (@rmjarvis can comment more).

We don't have production code to handle streaks, although we are of course aware of them during stacking.  There are some prototypes floating around HSC.

**Comment by sethdigel** [comment]
> 1. Scattered light from bright stars. Are reflections off of the CCDs and internal camera optics should be included by phosim. I think this means that we expect to see ghosting. Does DM have any mechanism to flag/remove affected CCDs?

The input catalogs of stars for the simulations are modified to limit the brightest stars to magnitude 10.  I think that this was done primarily to speed up the simulations; bright stars had a non-linear impact on overall CPU time.  Ghosting is still present, I'm sure, but the effects are not at a level that it will be in real life.

**Comment by RobertLuptonTheGood** [comment]
Thanks, Seth.  That makes these simulations uninteresting for understanding ghosts (and I don't know if there are ghoulies -- things like scattering off exposed bolt heads)

**Comment by kadrlica** [comment]
Thanks Seth. @rmandelb can we get these items added to your table? We may also want to place the table in a more permanent/accessible location (e.g., the sims specifications document?).

**Comment by rmjarvis** [comment]
@kadrlica psst.  Anyone can edit the table...  :)

**Comment by rmandelb** [comment]
@kadrlica - I agree the table needs to live somewhere more easily findable, but not just yet... I think we need to iterate more in this issue and then once we feel it's stable-ish we can put it in the sims specification document or some other sensible place.

I give 100% free permission for anybody to edit my comment above with the table in it. (I don't normally edit others' comments but in this case, please, have at it!  I see that Mike and Chris have been doing so and anybody else is welcome.).  With that said, I am actually thinking the two (e) columns are not quite what we want.  Here's why:

It is not that informative to say things like "validated against lab data for ITL sensors" or whatever.  This is not the level of information that I or others who care about the detector effects really care about, and I realize I may have been vague about this before.  But what I would find actually useful is as follows: we know that people have been characterizing these sensors for a while, in a variety of ways, and our understanding of the effects we have mentioned has evolved with time.  So which lab data?  From when?  Those are both relevant.  Moreover, nobody is directly inserting lab data as-is into their sims.  In general, people are fitting some kind of model to lab data, deciding whether that model describes the data well, and then putting the model into the sims.  So what's the model?  How well does it describe the lab data?  Again, that level of nuance is relevant.  If somebody says "This is based on ITL sensor tests that Andrei et al have been doing at BNL within the past year.  We used measurements of quantity X, and fit it to a sum of N Gaussians, increasing N until we could match X to within 97%." and points me to a plot in a paper / DESC Note / presentation then I have some idea what's going on.  Sorry for my original vagueness but after hearing the responses in this thread, I do think this kind of few-sentence high-level status is relevant.  And we're not putting a 3-sentence description of each model in the table, so maybe we should just remove it and have a bit of text for each of these.

Let me give some examples of why I think this is relevant, both as WL person who cares about sensor effects for my science and as somebody who is supposed to represent the analysis WG's interests:

- Maybe PhoSim and ImSim both have models that are of sufficient realism to be interesting, but they are different models -- we need to know that.  That can be very interesting, because then we can plan to use the different sims as different realizations of reality, or we could do things like define mitigation schemes on just one simulation and see how well they work on the other simulation to get a sense of residual systematic errors due to modeling assumptions. 
- Or if both use 100% identical models, again, I think this is relevant info (because then we don't expect to learn more/less about this specific effect from one sim than the other, so it informs how we plan to do our DC2 work).
- Or if a model is based on early lab tests that have since been superceded by other lab tests that led to greater understanding, such that the way the effect is currently simulated is not actually useful for future studies of mitigation schemes, it's important to know that - because we might not want to turn that model on.  It's OK to not simulate every effect that we could possibly simulate, if it's not going to be informative in the way that we want it to be.

I'm not requesting this information instantly, since we are all busy people and it's not the highest priority that we have this *right now*, but I wanted to get it on the record that I think having this few-sentence level of information by the DC2 checkpoint would be helpful.

If this information is already available, then I apologize - and in that case I have egg on my face - but I looked in the DC2 planning document and couldn't find it.

Thoughts?

*Short summary*: let's keep updating the table in this thread, move it to the DC2 planning document or some other suitable place once it's converged, and I request a bit more information about these various effects.

**Comment by kadrlica** [comment]
@rmjarvis thanks. I have a similar perception of github etiquette as Rachel (i.e., I don't edit others' posts without explicit permission). I see that in this case it is given, and in fact encouraged.

**Comment by rmandelb** [comment]
Small follow-up to my very long (sorry) comment above:

I was just looking at the Run 1.1p Note, and it has a table of simulated detector effects.  For bleeding, hot columns, hot pixels, nonlinearity, and cosmic rays, there is a reference to the 2015 PhoSim paper with a mention of what section of the paper describes those, so perhaps I can find the kind of summary I was hoping for there (I will have to go read).  However, the table also indicates that for brighter/fatter, tree rings, and cross-talk, there has been some updates so that reference is not relevant.  

And obviously information about these effects is needed for ImSim as well (that's not in the Note because it was describing Run 1.1p).

**Comment by salmanhabib** [comment]
I am a little puzzled about how we go back and forth about detailed modeling of CCD effects and backgrounds etc. If super-detailed modeling of CCD effects is needed, that is not the remit of DC2 or test run 1.2, it requires a separate effort at the level of a chip or a raft, working closely with Lab data and eventually with data from the mountain. We do not have the personpower or the time or even the real motivation for such a detailed detector simulation program given that no one has even proposed actual metrics against which to work (see the paucity of remarks in #142). Indeed, unless we have a simplified reference run, I am willing to bet that people won't be able to define such metrics. It's only by seeing explicit failure modes that such metrics will eventually appear. What we need is a definition of "good enough for now" with which to proceed. This definition needs to come from the science working groups (vetted by DM who presumably understand the present state of their software). 

We need a solution to the halting problem.

**Comment by rmandelb** [comment]
@salmanhabib - First, let me reiterate that what I said is unrelated to Run 1.2.

I am not saying it's the responsibility of the people producing DC2 to provide super-detailed brand-new modeling of CCD effects.  Rather, what I am saying is that I want information about how the effects are modeled, and the information I need to understand them is a bit more detail than "based on LSST sensors" and rather less detail than, say, writing a paper.  The examples I gave above should demonstrate what I mean about the information I want. 

The point is that we need enough information to decide whether the way the effect is modeled is sufficient for our science.  This means we also need some more detailed plans for what science we'll do with the DC2 sims - which is why I had a chat with our technical coordinator yesterday about how we can get some more well-defined SAWG+analysis WG plans that would enable a more concrete statement of needs.

So given the better definition of what the working groups want to do in DC2, we can then take the information I requested (how are these effects included in the sims) and decide what to do: can we do that science with those models?  Or should we kick this effect down the road to DC3 or some non-DC simulation study?  But that does require more information: what is the science?  What's in the sims?  Currently, for some sensor effects, we have neither.  I am working on the first question and my message above is requesting information to answer the second.  For the first question, I suspect the answer is that detailed quantitative studies will be planned for a very few limited effects and the rest not.

For example, if all these groups want to do with tree rings is get an order of magnitude estimate of whether they impact the galaxy positions/photometry in a measurable way, then it could be that the response to learning "the tree ring model is based on 2015 lab data and there's new data we're using now to build a new model" is "We don't care to wait on that new model, because they're not trying to do anything really quantitative with it".  However, if their goal is something much more quantitative like "we want to know whether residual brighter/fatter effect leads to a bias in WL estimates of w that exceeds statistical error", then we want to know we're using the latest LSST camera-team models.  If they are not sure what they will do with some effect, then I want to at least be able to tell them how the effect is simulated, in case that leads to further thought/questions on their part (again, at the level of information from my above message). 

And finally: the onus is not entirely on us to validate these things.  There is an LSST camera team, we have a technical coordinator and a DESC liaison to the camera team, and we can ask "What is the latest Project-adopted model for effect X and have you checked how these simulation tools (which they do use!) reproduce it?"  If our camera team liaison says "We've been using PhoSim v3.7 to simulate brighter/fatter, and it matches our lab data from BNL in 2017 at this level - here's a plot" -- then great. If a group really cares about the details of the sensor effect then it would be better anyway to make sure we are fully in sync with the Project.  (And again, I expect that would be the case for a very limited number of effects.)

**Comment by salmanhabib** [comment]
@rmandelb , I only partially agree with what you are saying. To get things started right now we only need to be sufficiently close to the right answer. If a certain tree ring model gives data close enough to that from the lab, (and let's say the effect is small enough to be correctable by flats as @RobertLuptonTheGood noted), then why does it matter whether the science WG understands how this was done? There's plenty of other things they don't understand. For example, they only understand in principle how the galaxy catalogs were created, the important thing they should care about, at first order, is that validation tests are passed. 

As to the LSST camera team, nothing has stopped the project from giving us information and of course I would welcome such information. But they do things for their own reasons and it may be that many of our requirements are much stronger than theirs. BTW, who is our camera liaison and why have they not said anything useful so far?

My complaint is that we are putting the cart before the horse. What's the point of worrying about models (second-order) when we don't even have first-order validation criteria?

**Comment by rmandelb** [comment]
> If a certain tree ring model gives data close enough to that from the lab, (and let's say the effect is 
> small enough to be correctable by flats as @RobertLuptonTheGood noted),

Well - actually that's my point, and I think we do agree. What I'm saying is that I want that few-sentence description of what is the model (which by definition must be pretty high-level) but mostly how the model was validated / how does it compare with lab data.  That was precisely my example: "This is based on ITL sensor tests that Andrei et al have been doing at BNL within the past year. We used measurements of quantity X, and fit it to a sum of N Gaussians, increasing N until we could match X to within 97%."   And maybe a plot. I want to be able to say the basics of what the model is doing and how was it validated and against what data.  Note, this validation is most likely not something we're doing with test runs, but rather it's validation that was done by whoever implemented the sensor model. 

To take your other excellent example, which is one that I also agree with: I have no doubt that if you ask a random WL person who paid attention to the last DC2 telecon, they could tell you the high-level summary of what Andrew Hearin has done, but not most of the details - but they can say that he validated his model against SDSS and DEEP2 clustering data within some redshift range, and it matched at the X% level.  They saw plots of these things and had the opportunity to ask questions about them.  If they look at other DESCQA tests they can get some other examples of how it does / doesn't match data, and at what level.  This is sufficient information for most purposes: the basics of the model, what data was it validated against and with what criteria.  (So moving back to the sensors example, it's more detail than "it's a model that matches LSST sensors" and it's less than, say, a long technical paper.)  I haven't seen that information for some of these effects.  Speaking as a weak lensers who cares about sensor effects and image simulations: I have sometimes used simulations to do science where I didn't understand every detail of the algorithm for some sensor effect, but I have never used simulations to do science where I didn't have the above type of information.

Steve Ritz is our camera liaison and he has reached out as a result of this discussion.

**Comment by smr456** [comment]
Hi Salman,
  I am the Camera Liaison and Pierre is the technical coordinator. I had not read this thread until yesterday and I thought it would be best to talk with individuals first, which I am doing, before adding to this long (29 and counting) thread, to try to avoid further misunderstandings.
Regards, Steve

**Comment by salmanhabib** [comment]
@rmandelb Ok, cool, so we are on the same page then. I have no doubt that at the level of description you are talking about we can have everything under control. So all we need now is an agreement on the validation datasets (here @smr456 can provide valuable information, thanks for joining this thread, Steve!) and the validation criteria (various WGs).

We are working with BNL to figure out the PhoSim parameters using their lab data; hopefully will have results to share soon. Meantime, those interested should check out the Park et al paper:

http://iopscience.iop.org/article/10.1088/1748-0221/12/05/C05015/pdf

**Comment by rmandelb** [comment]
@salmanhabib - Thanks.  Re: validation criteria, I am working with Pierre Antilogus to get SAWG and analysis WGs talking a bit more about project definition, which is necessary to provide validation criteria.  But it's likely to be in the spirit of "we're not doing detailed tests of the impact of X/Y/Z, so as long as they agree with data to some relatively low tolerance level (factors of a few) it's fine.  We are doing detailed tests of the impact of these two effects, so there we need some more stringent validation criteria against the most recent lab data."  And then the information we receive about these effects will still be useful, even though we are going to be more discriminating about some effects than others.

Thank you for the paper.  I gather that what's shown there is data against which the sims are being validated for one specific effect (tree rings).

Taking a cue from Steve R, I think some offline discussion between the people who've been referenced here would be useful to take this a bit further and get on the same page.

**Comment by salmanhabib** [comment]
@rmandelb Yes, yes, yes, all I want is what you mention above -- how well do we have to do to be "good enough for now". 

Is there a project data repository that can be used to set up a baseline dataset for the validation process? Perhaps @smr456 and @PierreAstier can help with this.

**Comment by antilogus** [comment]
Just to comment/check ( if you say nothing , it's that you agree ;-) ... haii large trafic expected ) : 
- size of the CCD effects are / will be check in contact with SAWG / camera ( @nomerot  and @PierreAstier  are in the loop ) ... and if there is still doubt on some of them  as we start DC2 : we should not put them in.
- and sure , including exact lab results will be great , but the main issue that we will face , will not be the exact size of the effect put in , but the size of what will be left over a DM processing ...
- and here 2 remarks : 
 --- it's  a chicken and egg issue : right now in the table above there is indeed "y" on the future DM implementation of the # corrections ...but for most of them the corrections, if any,  are not "perfect" today and we can expect that the effort will be put where it makes most sens ... but the full process is on going , and will take times ...( related example  : tree ring correction for e2v will probably never exist) ... and I'm not convinced that all effects listed in the table above , setup with realistic value , are "a"  (harmless) .
 ---  beyond the current capability of DM to correct these effects , as @RobertLuptonTheGood  underlined it , we should be sure that DM will have all the dataset to compute the corresponding correction ( ex : flats at # flux ... # wavelength for some effect ( edge effect ) ...) ... and associated manpower :  the estimation  of some corrections  will request important dedicated effort ( remark : today  DM probably will better put its effort to tune linearity correction from test lab , than from simulation ... )  I think this deserve  an other column in the table  : which dataset is needed , and are we sure that the effort to compute the corrections from these dataset is available ? 
- from the above remarks , and from a pure DESC point of view , we should I think consider in some case to add "smaller effect " than the real one  in DC2 , not to inject the effect as it is expected , but the effect at a "reasonable post DM " scale ...this will grant for some effect that they are of "a" type ( no damage ) , and still allow to test their impact.  ( linearity ? xtalk  ? ) 
- and for  "new" effect like BF ,  for which we are missing practice / knowledge/  enough knowledge for full software validation  , we should think twice on what we want to do with DC2 :  showing that DM can correct a simulation cannot be our goal (we have lab data for that ) ... the goal should be to evaluate the impact of a residual BF to the science . The size and the shape of this residual are in question ... in particular the shape of the residual can be  quite different than the shape of the effect itself ... In this case I'll be more in favor to consider doing DC2-post processing to study the BF impact : it could mean to add on demand residual BF to the  images  (we can provide this from corrected test lab data , or pure guess )  prior a standard DM run  ( not correcting for BF) .  I know this is not a "low level simulation" , but could be much more pragmatic  than putting large effort on correcting the simulation : we could directly test what residual matter to WL ... and then put the effort on the correction side .( The main question on this topic in SAWG today is : down to which level we should correct the effect ?)

**Comment by rmjarvis** [comment]
> if there is still doubt on some of them as we start DC2 : we should not put them in.

I definitely do not agree with this.  Unless the simulated effect is wildly different than reality (e.g. it seems the airglow variation in PhoSim), I think we should generally put in all the effects we think are going to be in the real data.  

... modulo the questions Rachel posed at the start of this thread.  I agree there is good reason not to put in proper motions.  But everything else mentioned here I think has a well-motivated simulation model, which will be either a good test of DM's ability to remove the effect or a good evaluation of how well we will need to have DM remove the effect down the line.  Both are useful results.

As I see it, the understanding of how well the models correspond to reality is more useful when we draw up conclusions from DC2.  For instance,

> We found that the photometric variation due to tree rings is significant at the level of X%, which leads to such and such bias on the red sequence identification in RedMapper.  The model used is expected to match the real LSST data, since it was verified by such and such a procedure.  This implies that this effect needs to be corrected at Y level of fidelity by DM.  Work on this front is proceeding.

Or 

> We found that the photometric calibration by tying together the zeropoints from overlapping images was sufficient for X Y and Z uses.  However, the airglow and cloud model in this simulation was not as complicated as expected in real data, so we cannot completely conclude that the existing algorithms are sufficient for LSST Year 1 analysis.  More simulations with realistic sky levels are required.

So it is important to have this information collected somewhere in sufficient detail to write these kinds of statements.  But I don't think we need to base our decision of whether or not to include something in DC2 on how precisely we think it matches reality.

**Comment by salmanhabib** [comment]
re @RobertLuptonTheGood I think we can check off the first requirement, and I believe also the second, if we are given explicit requirements about how to generate them.

**Comment by rmandelb** [comment]
Per discussion on DC2 telecon, an additional factor in the decision criteria for effects to include is this:

How expensive is it to produce the calibration products that are needed to correct for the effect?

For example, if including brighter/fatter in the sims would increase the needed number of flats by a factor that is problematic due to the needed computational resources to produce those flats, then we would need a stronger argument to include b/f in DC2.

So what's needed is information about (a) what calibration products are needed in the absence of b/f or other sensor effects, just for the basic flat-fielding in DM, and (b) how each additional sensor effect increases the needed number of calibration products.

I believe we agreed that @PierreAstier can provide this (@antilogus and @katrinheitmann can correct me if I've misunderstood).  Once we have a sense of this cost, we can factor it into the decisions.

**Comment by rmandelb** [comment]
As we agreed during the discussion on the [SAWG telecon](https://confluence.slac.stanford.edu/display/LSSTDESC/January+-+June+2018) this week, I have added a column to the [table of effects earlier in this thread](https://github.com/LSSTDESC/DC2_Repo/issues/144#issuecomment-374758455) to indicate the status of calibration products that would be needed to make a correction for each effect within DM.

Many rows just have a ? because I need somebody else to fill it in.  However, the following information is filled in because it came directly from the telecon:
- Tree rings are correctly represented in PhoSim flats.
- We still do not have a clear path to getting flats that could be used to correct for b/f with PhoSim on a reasonable timescale, according to reports from @salmanhabib .

As a result, the analysis WG that had most strongly requested that b/f be included in DC2 (WL) has now retracted that request, at least for DC2 PhoSim simulations, since the b/f problem would be unrealistically challenging.

Once results of Run 1.1i are in (including information about calibration products, which looks promising but must be confirmed) then we may decide to include b/f in the DC2 ImSim simulations, but this is still TBD.  Another option is to test the impact of b/f on WL using special-order simulations rather than DC2, which would relax the timeframe and avoid holding up all of DC2 just to solve the flats problem.

**Comment by katrinheitmann** [comment]
Also, they are in Run 1.2p -- is that too small?

**Comment by rearmstr** [comment]
I've updated the DM column in the table to reflect what DM can do currently.  I have labeled the brighter-fatter 'y' in the table even though it isn't ready right now.  It should be ready soon.  

I wasn't what clouds means exactly, maybe someone can clarify?

**Comment by rmandelb** [comment]
Hi @rearmstr - 

> I have labeled the brighter-fatter 'y' in the table even though it isn't ready right now and hasn't been 
> tested. It should be ready soon. Maybe this should be changed to 'n'?

My own thinking is that correction code should get a 'y' if it's been tested with a range of reasonable inputs and the following has been shown to be true:

- it runs / is stable (e.g., our processing won't be in danger of actually crashing due to this code being enabled)
- it doesn't give crazy results that will mess us up in some other way.

Presumably this would be in some DM unit tests or validation test suite?  And it sounds like the b/f correction code does not meet this standard.

With that said, this is just my thinking, and I would like to see whether others agree.

**Comment by rmjarvis** [comment]
Maybe there should be a label for "expected ready by the time DC2 images are produced"?  I.e. it doesn't meet Rachel's criteria for ready (which I think are very reasonable) yet, but we expect it to by the time we will need it.

**Comment by katrinheitmann** [comment]
Summary: This has been an extremely useful discussion to decide what to include in DC2. A summary of what we did in the end is given in the main DC2 paper. Since DC2 is now in full production, I close this issue.
